### PR TITLE
Pipeline configuration

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Added pipeline configuration files and translate rapid pro keys modules. Pending https://github.com/AfricasVoices/Project-ICRAF/pull/2, https://github.com/AfricasVoices/Project-ICRAF/pull/3 and https://github.com/AfricasVoices/Project-ICRAF/pull/4 . Project start and end dates to be confirmed. Future prs will be module based rather than batches.

### CODE REVIEW

File: src/libs/pipeline_configuration.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    AGE = _open_scheme("age.json")
+    LIVELIHOOD = _open_scheme("livelihood.json")
+    GENDER = _open_scheme("gender.json")
+    CONSTITUENCY = _open_scheme("constituency.json")
+    ORGANIZATIONS = _open_scheme("organizations.json")
+    CURRENT_PRACTICES = _open_scheme("current_practises.json")
+    NEW_PRACTICES = _open_scheme("new_practises.json")
+    REASONS_FOR_NEW_PRACTICES = _open_scheme("reasons_for_new_practices.json")
+    UPPER_TANA_PRACTISES = _open_scheme("upper_tana_practises.json")
+    YES_NO = _open_scheme("yes_no.json")
```
Comment (as2388) [inline src/libs/pipeline_configuration.py]: (obv. this needs to be updated once my comments on #4 are addressed)

File: src/libs/pipeline_configuration.py [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                    cleaner=None,
+                    code_scheme=CodeSchemes.NEW_PRACTICES),
+        
+        CodingPlan(raw_field="new_practices_reasons_raw",
+                    coded_field="new_practices_reasons_coded",
+                    time_field="new_practices_reasons_time",
+                    coda_filename="new_practices_reasons.json",
+                    analysis_file_key="new_practices_reasons_",
+                    cleaner=None,
+                    code_scheme=CodeSchemes.REASONS_FOR_NEW_PRACTICES),
```
Comment (as2388) [inline src/libs/pipeline_configuration.py]: Might be worth renaming this scheme to bring it in line with the field names.

File: src/libs/pipeline_configuration.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+                    time_field="livelihood_time",
+                    coda_filename="livelihood.json",
+                    analysis_file_key="livelihood",
+                    cleaner=None,
+                    code_scheme=CodeSchemes.LIVELIHOOD),
+    ]
+    
+    def __init__(self, rapid_pro_domain, rapid_pro_token_file_url, rapid_pro_key_remappings):
+        """
+        Arguments:
```
Comment (as2388) [inline src/libs/pipeline_configuration.py:228]: If you're keen to change the docstring format raise the issue on  #tech_data_team

File: configurations/pipeline_config.json [hunk 4]
```diff
@@ -1,4 +1,50 @@
...
+    {"RapidProKey": "Age (Value) - icraf_s01_demogs", "PipelineKey": "age_raw"},
+    {"RapidProKey": "Constituency (Value) - icraf_s01_demogs", "PipelineKey": "constituency_raw"},
+    {"RapidProKey": "Livelihood (Value) - icraf_s01_demogs", "PipelineKey": "livelihood_raw"},
+
+    {"RapidProKey": "Gender (Time) - icraf_s01_demogs", "PipelineKey": "gender_time"},
+    {"RapidProKey": "Age (Time) - icraf_s01_demogs", "PipelineKey": "age_time"},
+    {"RapidProKey": "Constituency (Time) - icraf_s01_demogs", "PipelineKey": "constituency_time"},
+    {"RapidProKey": "Livelihood (Time) - icraf_s01_demogs", "PipelineKey": "livelihood_time"},
+
+    {"RapidProKey": "Current_Practices (Run ID) - icraf_s01_follow_up_survey", "PipelineKey": "current_practices_run_id"},
```
Comment (as2388) [inline configurations/pipeline_config.json]: Why do you need the run id for follow-up surveys?

**Review by as2388 — APPROVED** [review]

File: src/libs/pipeline_configuration.py (hunk 1 above)
Comment (IsaackMwenda) [reply src/libs/pipeline_configuration.py]: Updated thanks!

File: src/libs/pipeline_configuration.py (hunk 2 above)
Comment (IsaackMwenda) [reply src/libs/pipeline_configuration.py]: Renamed to NEW_PRACTICES_CHALLENGES awaiting review from RDA

File: src/libs/pipeline_configuration.py (hunk 3 above)
Comment (IsaackMwenda) [reply src/libs/pipeline_configuration.py:228]: Will do thanks

File: configurations/pipeline_config.json (hunk 4 above)
Comment (IsaackMwenda) [reply configurations/pipeline_config.json]: Removed thanks

**Pull request merged**